### PR TITLE
[Bun] Add Bun runtime

### PR DESCRIPTION
**Pull request by marcbarbosa**
Initial (vanilla) implementation for [Bun](https://bun.sh): `/json` and `/plaintext`

Reference: https://github.com/TechEmpower/FrameworkBenchmarks/issues/7646

### CODE REVIEW

File: frameworks/TypeScript/bun/bun.dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM oven/bun
```
Comment (nbrady-techempower) [inline frameworks/TypeScript/bun/bun.dockerfile:1]: Could you lock this down to a specific version please

Comment (marcbarbosa) [reply frameworks/TypeScript/bun/bun.dockerfile]: Done

File: frameworks/TypeScript/bun/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+type HandlerFn = (req?: Request) => Promise<Response> | Response;
+
+const options: ResponseInit = { headers: { "Server": "Bun" } };
+
+const handlers = new Map<string, HandlerFn>();
+handlers.set("/json", () => Response.json({ message: "Hello, World!" }, options));
+handlers.set("/plaintext", () => new Response("Hello, World!", options));
```
Comment (WebReflection) [inline frameworks/TypeScript/bun/src/index.ts:7]: out of curiosity, shouldn't this be `Response.text("Hello, World!", options)` instead?

File: frameworks/TypeScript/bun/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+const server = Bun.serve({
+  port: 8080,
+  fetch(req: Request) {
+    return handlers.get(new URL(req.url).pathname)?.call(req);
```
Comment (WebReflection) [inline frameworks/TypeScript/bun/src/index.ts:12]: as a benchmark, I am not sure this it idiomatic or well performing + the `call(req)` when functions are arrows look off ..

```js
fetch: (req: Request) => handlers.get(new URL(req.url).pathname)?.()
```

looks already better, but I am not sure that's how people really write routes in Bun neither .. if it's idiomatic benchrmaks we're after, this PR doesn't look like the best we can do ... thoughts?

File: frameworks/TypeScript/bun/src/index.ts (hunk 2 above)
Comment (marcbarbosa) [reply frameworks/TypeScript/bun/src/index.ts]: > out of curiosity, shouldn't this be Response.text("Hello, World!", options) instead?

Not really, `new Response()` returns `text/plain` as content type already, required by the `/plaintext` scenario.

> or better, what is Response.json in there?

`Response.json` returns the object serialized and `application/json` as content type, required by the `/json` scenario.

File: frameworks/TypeScript/bun/src/index.ts (hunk 3 above)
Comment (marcbarbosa) [reply frameworks/TypeScript/bun/src/index.ts]: I agree, actually I'm refactoring my PR to simplify this part and not implementing anything other than what is strictly necessary. Just plain `if` statements.

File: frameworks/TypeScript/bun/src/index.ts (hunk 2 above)
Comment (WebReflection) [reply frameworks/TypeScript/bun/src/index.ts]: not in the docs and not standard though, maybe a comment would do

Comment (marcbarbosa) [reply frameworks/TypeScript/bun/src/index.ts]: This is the documentation for `Response.json`: https://developer.mozilla.org/en-US/docs/Web/API/Response/json_static

And I believe `text/plain` is the default content type.

Do you think an extra comment is really needed?

Comment (WebReflection) [reply frameworks/TypeScript/bun/src/index.ts]: I’ve missed that … my bad. No comment needed

**Comment by nbrady-techempower** [comment]
@marcbarbosa Let me know when you're done refactoring. This looks ready.

**Comment by marcbarbosa** [comment]
@nbrady-techempower Done

**Comment by Jarred-Sumner** [comment]
This needs parallelism to be potentially competitive with native languages — you can use Worker and enable reusePort. request.url already runs the URL parser so you’re running it twice here

**Pull request merged**